pkg/repositories/gormimpl/dataset_test.go

```diff
@@ -88,6 +88,36 @@ func TestGetDataset(t *testing.T) {
 	assert.Equal(t, dataset.Version, actualDataset.Version)
 }
 
+func TestGetDatasetWithUUID(t *testing.T) {
+	dataset := models.Dataset{
+		DatasetKey: models.DatasetKey{
+			UUID: "test-uuid",
+		},
+	}
+	expectedResponse := make([]map[string]interface{}, 0)
+	sampleDataset := make(map[string]interface{})
+	sampleDataset["project"] = dataset.Project
+	sampleDataset["domain"] = dataset.Domain
+	sampleDataset["name"] = dataset.Name
+	sampleDataset["version"] = dataset.Version
+
+	expectedResponse = append(expectedResponse, sampleDataset)
+
+	GlobalMock := mocket.Catcher.Reset()
+	GlobalMock.Logging = true
+
+	// Only match on queries that append expected filters
+	GlobalMock.NewMock().WithQuery(`SELECT * FROM "datasets"  WHERE "datasets"."deleted_at" IS NULL AND (("datasets"."uuid" = test-uuid)) ORDER BY "datasets"."project" ASC LIMIT 1`).WithReply(expectedResponse)
```

Comment:
Doesn't the actual service endpoint need logic that does this?  I'm probably missing context, but seems like there should be some higher level abstraction to test.

Reply:
Yep totally, that will come in a future PR. 

After making this change I will have another PR that fixes the foreign keys of the `ArtifactData` and `Tag` table to use DatasetUUID instead of `Project, Domain, Name, Version` and also allow service queries to query Datasets by UUID.